src/Provider/AbstractProvider.php

```diff
@@ -344,13 +353,8 @@ public function getAccessToken($grant = 'authorization_code', array $params = []
      */
     public function getAuthenticatedRequest($method, $url, AccessToken $token)
     {
-        $factory = $this->getHttpClient()
-            ->getConfiguration()
-            ->getMessageFactory();
-
-        $request = $factory->createRequest($url, $method);
-        $request->addHeaders($this->getHeaders($token));
-        return $request;
+        $options = ['headers' => $this->getHeaders($token)];
+        return $this->getGuzzleRequest($method, $url, $options);
```

Comment:
And here we have a very convenient method that gives us back an authenticated request, also configured (privately) by Guzzle.